inc/Engine/Optimization/RUCSS/ServiceProvider.php

```diff
@@ -36,10 +41,34 @@ public function register() {
 			->withArgument( $this->getContainer()->get( 'options' ) );
 		// Instantiate the RUCSS Resources Table class.
 		$this->getContainer()->add( 'rucss_resources_table', 'WP_Rocket\Engine\Optimization\RUCSS\Database\Tables\Resources' );
+		$this->getContainer()->add( 'rucss_resources_query', 'WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery' );
 		$this->getContainer()->add( 'rucss_database', 'WP_Rocket\Engine\Optimization\RUCSS\Admin\Database' )
 			->withArgument( $this->getContainer()->get( 'rucss_resources_table' ) );
 		$this->getContainer()->share( 'rucss_admin_subscriber', 'WP_Rocket\Engine\Optimization\RUCSS\Admin\Subscriber' )
 			->withArgument( $this->getContainer()->get( 'rucss_settings' ) )
 			->withArgument( $this->getContainer()->get( 'rucss_database' ) );
+
+		$this->getContainer()->add( 'local_cache', '\WP_Rocket\Engine\Optimization\AssetsLocalCache' )
+			->withArgument( rocket_get_constant( 'WP_ROCKET_MINIFY_CACHE_PATH' ) )
+			->withArgument( rocket_direct_filesystem() );
```

Comment:
This is already loaded elsewhere, why the need to re-add it here?

Reply:
it's called using `new` not via the container here:

https://github.com/wp-media/wp-rocket/blob/f92eb9edbb0dd8710061d1643da9b5f6745d56b4/inc/Engine/Optimization/Minify/JS/Subscriber.php#L47

https://github.com/wp-media/wp-rocket/blob/f52d197c1da902b1c8b065340f7bff95853566a5/inc/Engine/Optimization/Minify/CSS/Subscriber.php#L45

this is why I created it here, do u think we need to refactor this code to use the container

Reply:
Not for now, it's out of scope of what we're doing here.

Reply:
I vote for reusing everywhere from container.